Mathlib/Topology/Separation.lean

```diff
@@ -2286,51 +2299,91 @@ instance (priority := 100) NormalSpace.of_compactSpace_r1Space [CompactSpace X]
     NormalSpace X where
   normal _s _t hs ht := .of_isCompact_isCompact_isClosed hs.isCompact ht.isCompact ht
 
-/-- A regular topological space with second countable topology is a normal space.
+/-- Lemma used to prove that a regular topological space with Lindelöf topology is a normal space,
+and (todo) a perfectly normal space is a completely normal space. -/
+lemma countable_covers_witnessing_separated_nhds {h k : Set X}
+    (h_cov: ∃ u : ℕ → Set X, h ⊆ ⋃ n, u n ∧
+      ∀ n, IsOpen (u n) ∧ Disjoint (closure (u n)) k)
+    (k_cov: ∃ u : ℕ → Set X, k ⊆ ⋃ n, u n ∧
+      ∀ n, IsOpen (u n) ∧ Disjoint (closure (u n)) h) : SeparatedNhds h k := by
+  rcases h_cov with ⟨u, u_cov, u_props⟩
+  rcases k_cov with ⟨v, v_cov, v_props⟩
+  use ⋃ n : ℕ, u n \ (closure (⋃ m ∈ {m | m ≤ n}, v m))
+  use ⋃ n : ℕ, v n \ (closure (⋃ m ∈ {m | m ≤ n}, u m))
+  have open_lemma : ∀ (u₀ a : ℕ → Set X), (∀ n, IsOpen (u₀ n)) →
+    IsOpen (⋃ n, u₀ n \ closure (a n)) := fun _ _ u₀i_open ↦
+    isOpen_iUnion (fun i ↦ IsOpen.sdiff (u₀i_open i) (isClosed_closure))
+  have cover_lemma : ∀ (h₀ : Set X) (u₀ v₀ : ℕ → Set X),
```

Comment:
Don't these two open/cover lemma really deserve any independent status that would make it worth to single them out of the proof? In other words, would you expect them to be potentielly useful in other proofs?

Reply:
For `open_lemma`, that seems possible. I'd thought about it but wasn't sure where the threshold was.

Maybe `cover_lemma` too. I'll stew on it.

Reply:
I think the necessity of this will be clear as I work on #13517, and I think it's best to defer this to that PR.